src/lsm/compaction.zig

```diff
@@ -287,6 +301,7 @@ pub fn CompactionType(
             assert(compaction.status == .processing);
             assert(compaction.callback != null);
             assert(!compaction.merge_done);
+            assert(compaction.strategy == .table_merge);
```

Comment:
For example, on line 303 we say only `merge_done` but then on line 304 we say `table_merge`.

Seems like the enum could be just `.merge` and it would reuse the "human grammer", the reader's internal parsing that they just did on line 303.